Framework/Algorithms/src/GetEiMonDet2.cpp

```diff
@@ -425,7 +425,11 @@ void GetEiMonDet2::parseIndices(std::vector<size_t> &detectorIndices,
         spectraListParser.parse(getProperty(PropertyNames::DETECTORS)));
     auto back = std::back_inserter(detectorIndices);
     std::copy(detectors.begin(), detectors.end(), back);
-    monitorIndex = getProperty(PropertyNames::MONITOR);
+    int monitor = getProperty(PropertyNames::MONITOR);
+    if (monitor < 0) {
```

Comment:
Yes, of course. Somehow I was thinking that somebody might be mad enough to use negative detector id or spectrum number here. I guess that will never happen and validating the *Monitor* property makes the check here superfluous.